doc/design.adoc

```diff
@@ -0,0 +1,637 @@
+
+| `externalService`
+| External service that is accessible from other sites.
+| https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#service-v1-core[coreV1.Service]
+| true
+|
+
+|=======================
+
+
+[[infinispanremotesitespec]]
+#### `InfinispanRemoteSiteSpec`
+
+`InfinispanRemoteSiteSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `name`
+| Name of remote site.
+| string
+| true
+|
+
+| `type`
+| Type of remote site configuration.
+| `Static` or `Dynamic`
+| true
+|
+
+| `host`
+| Remote site host name.
+| string
+| true
+|
+
+| `port`
+| Remote site host port (only for `Static` type).
+| int32
+| false
+|
+
+| `secret`
+| Secret to connect to remote site (only for `Dynamic` type).
+| <<infinispansecretspec>>
+| false
+|
+
+|=======================
+
+
+[[infinispansecretspec]]
+##### `InfinispanSecretSpec`
+
+`InfinispanSecretSpec` defines how `InfinispanSpec` secrets are configured.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `type`
+| Type of secret.
+| `Credentials`, `Keystore` or `Token`
+| true
+|
+
+| `secretName`
+| Name of referenced secret.
+| string
+| true
+|
+
+|=======================
+
+If type is `Credentials`, Secret` is expected to contain username and password credentials.
+These would be defined in `stringData/username` and `stringData/password` fields respectively.
+
+If type is `Keystore`, `Secret` is expected to contain base64 encoded data in `data/keystore.p12` field.
+Optional keystore password would be located in `stringData/password` field.
+
+If type is `Token`, `Secret` is expected to contain base64 encoded data in `stringData/token` field.
+
+
+[[infinispanstatus]]
+### `InfinispanStatus`
+
+`InfinispanStatus` is the most recent observed status of the `InfinispanSpec`. Read-only.
+
+This is a proposal. TODO: Please comment
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `nodes`
+| Status of the nodes.
+| []<<InfinispanNodeStatus>>
+| true
+
+| `summary`
+| Short description of the cluster status. Now summary is `Ready` if all the pod conditions of type `Ready` are true 
+| string
+| true
+
+|=======================
+
+
+[[nodes]]
+#### `InfinispanNodeStatus`
+
+`InfinispanNodeStatus` is the most recent observed status of a pod running `InfinispanSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `name`
+| Name of the Pod.
+| string
+| true
+
+| `clusterMembers`
+| List of cluster members (in Infinispan CLI format)
+| string
+| true
+
+| `clusterSize`
+| Number of cluster members
+| int
+| true
+
+|=======================
+
+## Full Example
+
+.full-example.yaml
+[source,yaml]
+----
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: full-example-infinispan
+spec:
+  evictionPolicy: Reject
```

Comment:
Yeah, makes sense. I'm however not sure how a user should select whether to use a cache or a datagrid. I mean, there's shouldn't need to specify `cache.evictionPolicy` or `cache.replicationFactor` for that. Maybe just a `spec.profile` with options `cache|datagrid|dev` would work? `dev` would be for mode where everything is unsecured.

Reply:
I wanted to group this two parameters as they are tightly related since both are default cache configuration specific. (I mean cache configuration not the service profile). It would be nicer if we decided to add another cache specific params in future. (maybe `defaultCache` if not `cache`?):

```
defaultCache:
  replicationFactor: 3
  evictionPolicy: Reject
```
On a similar topic. Do we want to enable several cache specification in operator yaml for users that don't want to create caches dynamically (if there are any)? Having this spec in yaml and processing it through operator would be simpler and nicer compared to env vars processed by scripts.
```
caches:
- name: default
  replicationFactor: 3
  evictionPolicy: Reject
- name: custom
  replicationFactor: 2
  evictionPolicy: Evict
```